lib/osf-components/addon/components/file-actions-menu/delete-modal/component.ts

```diff
@@ -0,0 +1,33 @@
+import { inject as service } from '@ember/service';
+import { waitFor } from '@ember/test-waiters';
+import Component from '@glimmer/component';
+import { task } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
+import IntlService from 'ember-intl/services/intl';
+import File from 'ember-osf-web/packages/files/file';
+import captureException, { getApiErrorMessage } from 'ember-osf-web/utils/capture-exception';
+
+interface Args {
+    file: File;
+    closeModal: () => {};
+    onDelete: () => {};
+}
+
+export default class DeleteModal extends Component<Args> {
```

Comment:
The only problem with using the `<DeleteButton>` component is that the `<BsModal>` component is invoked where the `<DeleteButton>` is invoked. In this case it would be the dropdown menu. Clicking within the modal counts as a click-ouside of the dropdown menu, which will close the dropdown, and then the modal will also be destroyed.

Reply:
Gah, I see. That's unfortunate :/